src/index.js

```diff
@@ -1,217 +1,41 @@
-      documentDeclaration,
-      logPrefix,
-    });
-  }
-
-  async fetch({ location, serviceId, type, logPrefix }) {
-    console.log(`${logPrefix} Fetch '${location}'.`);
-    try {
-      return await fetch(location);
-    } catch (error) {
-      console.error(`${logPrefix} Could not fetch location: ${error}`);
-      this.emit('documentFetchError', serviceId, type, error);
-    }
-  }
-
-  /* eslint-disable class-methods-use-this */
-  async recordSnapshot({ content, serviceId, type, logPrefix }) {
-    const { id: snapshotId, path: snapshotPath } = await history.recordSnapshot(serviceId, type, content);
-
-    if (!snapshotId) {
-      return console.log(`${logPrefix} No changes, did not record snapshot.`);
-    }
-
-    console.log(`${logPrefix} Recorded snapshot in ${snapshotPath} with id ${snapshotId}.`);
-    return snapshotId;
-  }
-  /* eslint-enable class-methods-use-this */
-
-  async recordRefilter({ snapshotContent, snapshotId, serviceId, documentDeclaration, logPrefix }) {
-    const { type } = documentDeclaration;
-    const document = await filter(snapshotContent, documentDeclaration, this._serviceDeclarations[serviceId].filters);
-
-    const { id: versionId, path: documentPath } = await history.recordRefilter(serviceId, type, document, snapshotId);
-    if (versionId) {
-      console.log(`${logPrefix} Recorded version in ${documentPath} with id ${versionId}.`);
-      this.emit('versionRecorded', serviceId, type, versionId);
-    } else {
-      console.log(`${logPrefix} No changes after filtering, did not record version.`);
-    }
-  }
+    if (schedule) {
+      const rule = new scheduler.RecurrenceRule();
+      rule.minute = 30; // at minute 30 past every hour.
 
-  async recordVersion({ snapshotContent, snapshotId, serviceId, documentDeclaration, logPrefix }) {
-    const { type } = documentDeclaration;
-    const document = await filter(snapshotContent, documentDeclaration, this._serviceDeclarations[serviceId].filters);
+      const notifier = new Notifier(app.serviceDeclarations);
+      await app.attach(notifier);
 
-    const { id: versionId, path: documentPath, isFirstRecord } = await history.recordVersion(serviceId, type, document, snapshotId);
-    if (versionId) {
-      const message = isFirstRecord
-        ? `${logPrefix} First version recorded in ${documentPath} with id ${versionId}.`
-        : `${logPrefix} Recorded version in ${documentPath} with id ${versionId}.`;
-      console.log(message);
-      this.emit(isFirstRecord ? 'documentAdded' : 'versionRecorded', serviceId, type, versionId);
+      console.log('The scheduler is running…');
+      console.log('Documents will be tracked at minute 30 past every hour.');
+      scheduler.scheduleJob(rule, () => {
+        app.trackChanges(serviceId);
+      });
     } else {
-      console.log(`${logPrefix} No changes after filtering, did not record version.`);
+      await app.trackChanges(serviceId);
     }
+  } catch (error) {
+    console.error(error);
```

Comment:
Why don't we want to raise an exception and exit with a proper failure exit code?